unit-tests/func/extrinsics/test-extrinsics.cpp

```diff
@@ -0,0 +1,205 @@
+    pos_rot.pos_and_rot[0][2] = static_cast<double>(rotation[2]);
+
+    pos_rot.pos_and_rot[1][0] = static_cast<double>(rotation[3]);
+    pos_rot.pos_and_rot[1][1] = static_cast<double>(rotation[4]);
+    pos_rot.pos_and_rot[1][2] = static_cast<double>(rotation[5]);
+
+    pos_rot.pos_and_rot[2][0] = static_cast<double>(rotation[6]);
+    pos_rot.pos_and_rot[2][1] = static_cast<double>(rotation[7]);
+    pos_rot.pos_and_rot[2][2] = static_cast<double>(rotation[8]);
+
+    pos_rot.pos_and_rot[3][0] = 0.0;
+    pos_rot.pos_and_rot[3][1] = 0.0;
+    pos_rot.pos_and_rot[3][2] = 0.0;
+
+    pos_rot.pos_and_rot[0][3] = static_cast<double>(position[0]);
+    pos_rot.pos_and_rot[1][3] = static_cast<double>(position[1]);
+    pos_rot.pos_and_rot[2][3] = static_cast<double>(position[2]);
+
+    pos_rot.pos_and_rot[3][3] = 1.0;
+
+    return pos_rot;
+}
+
+
+
+// checking the extrinsics graph
+// steps are:
+// 1. get all the profiles
+// 2. for each pair of these profiles: 
+//    check that:
+//    point P(x,y,z,head,pitch,roll) in profile A transformed to profile B coordinates, and then transformed back to profile A coordinates == same point P
+TEST_CASE("Extrinsics graph - matrices 4x4", "[live]")
+{
+    rs2::context ctx;
+    auto list = ctx.query_devices();
+
+    for (auto&& device : list)
+    {
+        std::cout << "device: " << device.get_info(RS2_CAMERA_INFO_NAME) << std::endl;
+        auto sensors = device.query_sensors();
+
+        // getting stream profiles from all sensors
+        std::vector<rs2::stream_profile> profiles;
+        for (auto&& sensor : sensors)
+        {
+            std::vector<rs2::stream_profile> stream_profiles = sensor.get_stream_profiles();
+            for (auto&& profile : stream_profiles)
+            {
+                profiles.push_back(profile);
+            }
+        }
+
+        float start_point[3] = { 1.f, 2.f, 3.f };
+
+        for (int i = 0; i < profiles.size() - 2; ++i)
+        {
+            for (int j = i + 1; j < profiles.size() - 1; ++j)
+            {
+                rs2_extrinsics extr_i_to_j = profiles[i].get_extrinsics_to(profiles[j]);
+                rs2_extrinsics extr_j_to_i = profiles[j].get_extrinsics_to(profiles[i]);
+
+                position_and_rotation pr_i_to_j = matrix_4_by_4_from_translation_and_rotation(extr_i_to_j.translation, extr_i_to_j.rotation);
+                position_and_rotation pr_j_to_i = matrix_4_by_4_from_translation_and_rotation(extr_j_to_i.translation, extr_j_to_i.rotation);
+
+                position_and_rotation product = pr_i_to_j * pr_j_to_i;
+                // checking that product of extrinsics from i to j with extrinsiscs from j to i is identity matrix
+                REQUIRE(product.is_identity());
+
+                // checking with API rs2_transform_point_to_point
+                float transformed_point[3];
+                rs2_transform_point_to_point(transformed_point, &extr_i_to_j, start_point);
```

Comment:
Check using double types internally to improve tolerance precision

Reply:
did not help